### DISCUSSION TITLE
tax_level = "genus", but families are plotted as well

### DISCUSSION THREAD

**Discussion by Arvdputt** [david-barnett/microViz#151] [Q&A] [answered]
So I want to make a barplot of the genera of the microbiome of a range of tissues, but when I set the tax_level = "Rank6" (which is genus level) it also plots certain bacterial families, which I do not want.
this is my code:

# Fix taxa for MicroViz
{subset_env <- subset_env %>%
  tax_fix(unknowns = c("g__Incertae Sedis", "g__endosymbionts", "g__uncultured")) %>%
  tax_fix(unknowns = c("f__Unknown Family")) %>%
  tax_fix(unknowns = c("f__Unknown Family Rank5"))}

# MicroViz barplot tissue wrapped (only LO, no replicates) 
{subset_env %>%
  comp_barplot(
    tax_level = "Rank6", n_taxa = 20, other_name = "Other",
    taxon_renamer = function(x) stringr::str_remove(x, " [ae]t rel."),
    palette = distinct_palette(n = 20, add = "grey90"),
    merge_other = FALSE, bar_outline_colour = "darkgrey"
  ) +
  coord_flip() +
  facet_wrap(~ sampled_tisseu, nrow = 1, scales = "free")}

**Comment by david-barnett** [comment, marked as answer]
If some of your taxa are unclassified (or ambiguously classified) at the genus level, (which is common and looks to be the case for you, given your tax fix code), then tax_fix will fill the missing info with the family names (if known).

Those are most likely the family names you are seeing on your plots. You could exclude them using tax_select with deselect =TRUE, but I doubt that would be a good idea. 

Hope this helps 🙂 
David

**Reply by Arvdputt** [reply]
# MicroViz barplot gametophytes 
{subset_gam <- subset_gam %>%
    tax_select(unknowns = c("g__Incertae Sedis", "g__endosymbionts", "g__uncultured"), deselect = T) %>%
    tax_select(unknowns = c("f__Unknown Family"), deselect = T) %>%
    tax_select(unknowns = c("f__Unknown Family Rank5"), deselect = T)
  
  subset_gam %>%
    comp_barplot(
      tax_level = "Rank6", n_taxa = 20, other_name = "Other",
      taxon_renamer = function(x) stringr::str_remove(x, " [ae]t rel."),
      palette = distinct_palette(n = 20, add = "grey90"),
      merge_other = FALSE, bar_outline_colour = "darkgrey"
    ) +
    coord_flip()
}

I am not that good in R
